# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2022/11
# use cupyx.scipy.fft rather than cupyx.scipy.fftpack.fft
diff -ru PyTom-1.0b/pytom/bin/average.py PyTom-1.0b_fix_cupy_fftpack/pytom/bin/average.py
--- PyTom-1.0b/pytom/bin/average.py	2022-10-26 12:28:36.000000000 +0200
+++ PyTom-1.0b_fix_cupy_fftpack/pytom/bin/average.py	2022-11-29 10:16:29.789846259 +0100
@@ -245,9 +245,9 @@
     import os
 
     try:
-        from cupyx.scipy.fftpack.fft import fftn as fftnP
-        from cupyx.scipy.fftpack.fft import ifftn as ifftnP
-        from cupyx.scipy.fftpack.fft import get_fft_plan
+        from cupyx.scipy.fft import fftn as fftnP
+        from cupyx.scipy.fft import ifftn as ifftnP
+        from cupyx.scipy.fft import get_fft_plan
     except ModuleNotFoundError:  # TODO go to 'from cupyx.scipy.fftpack import ...' once fully moved to cupy > 8.3
         from cupyx.scipy.fftpack import fftn as fftnP
         from cupyx.scipy.fftpack import ifftn as ifftnP
diff -ru PyTom-1.0b/pytom/bin/ctfCorrection.py PyTom-1.0b_fix_cupy_fftpack/pytom/bin/ctfCorrection.py
--- PyTom-1.0b/pytom/bin/ctfCorrection.py	2022-10-26 12:28:36.000000000 +0200
+++ PyTom-1.0b_fix_cupy_fftpack/pytom/bin/ctfCorrection.py	2022-11-29 10:16:29.756845450 +0100
@@ -1,8 +1,8 @@
 #!/usr/bin/env pytom
 
-from cupyx.scipy.fftpack.fft import get_fft_plan
-from cupyx.scipy.fftpack.fft import fftn as fftnP
-from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+from cupyx.scipy.fft import get_fft_plan
+from cupyx.scipy.fft import fftn as fftnP
+from cupyx.scipy.fft import ifftn as ifftnP
 from cupyx.scipy.ndimage import map_coordinates
 from pytom.gui.guiFunctions import datatype
 import os
diff -ru PyTom-1.0b/pytom/gpu/gpuStructures.py PyTom-1.0b_fix_cupy_fftpack/pytom/gpu/gpuStructures.py
--- PyTom-1.0b/pytom/gpu/gpuStructures.py	2022-10-26 12:28:36.000000000 +0200
+++ PyTom-1.0b_fix_cupy_fftpack/pytom/gpu/gpuStructures.py	2022-11-29 10:16:26.590767895 +0100
@@ -60,9 +60,9 @@
         from cupyx.scipy.ndimage import map_coordinates
 
         try:
-            from cupyx.scipy.fftpack.fft import get_fft_plan
-            from cupyx.scipy.fftpack.fft import fftn as fftnP
-            from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+            from cupyx.scipy.fft import get_fft_plan
+            from cupyx.scipy.fft import fftn as fftnP
+            from cupyx.scipy.fft import ifftn as ifftnP
         except ModuleNotFoundError:  # TODO go to 'from cupyx.scipy.fftpack import ...' once fully moved to cupy > 8.3
             from cupyx.scipy.fftpack import get_fft_plan
             from cupyx.scipy.fftpack import fftn as fftnP
@@ -298,9 +298,9 @@
         from pytom.gpu.kernels import argmax_text, meanStdv_text
 
         try:
-            from cupyx.scipy.fftpack.fft import get_fft_plan
-            from cupyx.scipy.fftpack.fft import fftn as fftnP
-            from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+            from cupyx.scipy.fft import get_fft_plan
+            from cupyx.scipy.fft import fftn as fftnP
+            from cupyx.scipy.fft import ifftn as ifftnP
         except ModuleNotFoundError:  # TODO go to 'from cupyx.scipy.fftpack import ...' once fully moved to cupy > 8.3
             from cupyx.scipy.fftpack import get_fft_plan
             from cupyx.scipy.fftpack import fftn as fftnP
@@ -800,9 +800,9 @@
         from cupyx.scipy.ndimage import map_coordinates
 
         try:
-            from cupyx.scipy.fftpack.fft import get_fft_plan
-            from cupyx.scipy.fftpack.fft import fftn as fftnP
-            from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+            from cupyx.scipy.fft import get_fft_plan
+            from cupyx.scipy.fft import fftn as fftnP
+            from cupyx.scipy.fft import ifftn as ifftnP
         except ModuleNotFoundError:  # TODO go to 'from cupyx.scipy.fftpack import ...' once fully moved to cupy > 8.3
             from cupyx.scipy.fftpack import get_fft_plan
             from cupyx.scipy.fftpack import fftn as fftnP
@@ -1044,9 +1044,9 @@
         from pytom.agnostic.tools import create_sphere
 
         try:
-            from cupyx.scipy.fftpack.fft import get_fft_plan
-            from cupyx.scipy.fftpack.fft import fftn as fftnP
-            from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+            from cupyx.scipy.fft import get_fft_plan
+            from cupyx.scipy.fft import fftn as fftnP
+            from cupyx.scipy.fft import ifftn as ifftnP
         except ModuleNotFoundError:  # TODO go to 'from cupyx.scipy.fftpack import ...' once fully moved to cupy > 8.3
             from cupyx.scipy.fftpack import get_fft_plan
             from cupyx.scipy.fftpack import fftn as fftnP
diff -ru PyTom-1.0b/pytom/gui/ctfCorrectionNew.py PyTom-1.0b_fix_cupy_fftpack/pytom/gui/ctfCorrectionNew.py
--- PyTom-1.0b/pytom/gui/ctfCorrectionNew.py	2022-10-26 12:28:36.000000000 +0200
+++ PyTom-1.0b_fix_cupy_fftpack/pytom/gui/ctfCorrectionNew.py	2022-11-29 10:16:26.522766229 +0100
@@ -1,6 +1,6 @@
-from cupyx.scipy.fftpack.fft import get_fft_plan
-from cupyx.scipy.fftpack.fft import fftn as fftnP
-from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+from cupyx.scipy.fft import get_fft_plan
+from cupyx.scipy.fft import fftn as fftnP
+from cupyx.scipy.fft import ifftn as ifftnP
 from cupyx.scipy.ndimage import map_coordinates
 from pytom.gui.guiFunctions import datatype
 import os
Only in PyTom-1.0b_fix_cupy_fftpack/pytom/gui: .ctfCorrectionNew.py.swp
diff -ru PyTom-1.0b/pytom/gui/ctfCorrection.py PyTom-1.0b_fix_cupy_fftpack/pytom/gui/ctfCorrection.py
--- PyTom-1.0b/pytom/gui/ctfCorrection.py	2022-10-26 12:28:36.000000000 +0200
+++ PyTom-1.0b_fix_cupy_fftpack/pytom/gui/ctfCorrection.py	2022-11-29 10:16:26.524766278 +0100
@@ -1,6 +1,6 @@
-from cupyx.scipy.fftpack.fft import get_fft_plan
-from cupyx.scipy.fftpack.fft import fftn as fftnP
-from cupyx.scipy.fftpack.fft import ifftn as ifftnP
+from cupyx.scipy.fft import get_fft_plan
+from cupyx.scipy.fft import fftn as fftnP
+from cupyx.scipy.fft import ifftn as ifftnP
 from cupyx.scipy.ndimage import map_coordinates
 from pytom.gui.guiFunctions import datatype
 import os
